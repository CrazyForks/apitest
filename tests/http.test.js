const { spwanTest } = require("./utils");

describe("http", () => {
  test("invaid params mismatch path", async () => {
    const { stdout, code } = await spwanTest("http/invaid-params-mismatch-path.jsona", ["--ci"]);
    expect(code).toEqual(1);
    expect(stdout).toMatchSnapshot();
  });
  test("invaid params prop type", async () => {
    const { stdout, code } = await spwanTest("http/invaid-params-prop-type.jsona", ["--ci"]);
    expect(code).toEqual(1);
    expect(stdout).toMatchSnapshot();
  });
  test("invaid params type", async () => {
    const { stdout, code } = await spwanTest("http/invaid-params-type.jsona", ["--ci"]);
    expect(code).toEqual(1);
    expect(stdout).toMatchSnapshot();
  });
  test("invalid headers prop type", async () => {
    const { stdout, code } = await spwanTest("http/invalid-headers-prop-type.jsona", ["--ci"]);
    expect(code).toEqual(1);
    expect(stdout).toMatchSnapshot();
  });
  test("invalid headers type", async () => {
    const { stdout, code } = await spwanTest("http/invalid-headers-type.jsona", ["--ci"]);
    expect(code).toEqual(1);
    expect(stdout).toMatchSnapshot();
  });
  test("invalid method type", async () => {
    const { stdout, code } = await spwanTest("http/invalid-method-type.jsona", ["--ci"]);
    expect(code).toEqual(1);
    expect(stdout).toMatchSnapshot();
  });
  test("invalid method value", async () => {
    const { stdout, code } = await spwanTest("http/invalid-method-value.jsona", ["--ci"]);
    expect(code).toEqual(1);
    expect(stdout).toMatchSnapshot();
  });
  test("invalid params miss", async () => {
    const { stdout, code } = await spwanTest("http/invalid-params-miss.jsona", ["--ci"]);
    expect(code).toEqual(1);
    expect(stdout).toMatchSnapshot();
  });
  test("invalid query prop type", async () => {
    const { stdout, code } = await spwanTest("http/invalid-query-prop-type.jsona", ["--ci"]);
    expect(code).toEqual(1);
    expect(stdout).toMatchSnapshot();
  });
  test("invalid query type", async () => {
    const { stdout, code } = await spwanTest("http/invalid-query-type.jsona", ["--ci"]);
    expect(code).toEqual(1);
    expect(stdout).toMatchSnapshot();
  });
  test("invalid req value", async () => {
    const { stdout, code } = await spwanTest("http/invalid-req-value.jsona", ["--ci"]);
    expect(code).toEqual(1);
    expect(stdout).toMatchSnapshot();
  });
  test("invalid res headers prop type", async () => {
    const { stdout, code } = await spwanTest("http/invalid-res-headers-prop-type.jsona", ["--ci"]);
    expect(code).toEqual(1);
    expect(stdout).toMatchSnapshot();
  });
  test("invalid res headers type", async () => {
    const { stdout, code } = await spwanTest("http/invalid-res-headers-type.jsona", ["--ci"]);
    expect(code).toEqual(1);
    expect(stdout).toMatchSnapshot();
  });
  test("invalid res status type", async () => {
    const { stdout, code } = await spwanTest("http/invalid-res-status-type.jsona", ["--ci"]);
    expect(code).toEqual(1);
    expect(stdout).toMatchSnapshot();
  });
  test("invalid res type", async () => {
    const { stdout, code } = await spwanTest("http/invalid-res-type.jsona", ["--ci"]);
    expect(code).toEqual(1);
    expect(stdout).toMatchSnapshot();
  });
  test("invalid url miss", async () => {
    const { stdout, code } = await spwanTest("http/invalid-url-miss.jsona", ["--ci"]);
    expect(code).toEqual(1);
    expect(stdout).toMatchSnapshot();
  });
  test("invalid url type", async () => {
    const { stdout, code } = await spwanTest("http/invalid-url-type.jsona", ["--ci"]);
    expect(code).toEqual(1);
    expect(stdout).toMatchSnapshot();
  });
  test("main", async () => {
    const { code } = await spwanTest("http", ["--ci"]);
    expect(code).toEqual(0);
  }, 60000);
});

describe("http trans", () => {
  test("no-check-trans", async () => {
    const { code } = await spwanTest("http/no-check-trans.jsona", ["--ci"]);
    expect(code).toEqual(0);
  }, 60000);
});

describe("http form", () => {
  test("form", async () => {
    const { code } = await spwanTest("http/form.jsona", ["--ci"]);
    expect(code).toEqual(0);
  }, 60000);
});

describe("http cookie", () => {
  test("cookie", async () => {
    const { code } = await spwanTest("http/cookie.jsona", ["--ci"]);
    expect(code).toEqual(0);
  }, 60000);
});
