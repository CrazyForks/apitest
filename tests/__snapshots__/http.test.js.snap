// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`http invaid params mismatch path 1`] = `
"main.test1.req.params: should match url params at line 5 col 15
"
`;

exports[`http invaid params prop type 1`] = `
"main.test1.req.params.id: should be scalar value at line 6 col 13
"
`;

exports[`http invaid params type 1`] = `
"main.test1.req.params: should be object value at line 5 col 15
"
`;

exports[`http invalid headers prop type 1`] = `
"main.test1.req.headers.key1: should be scalar value at line 6 col 15
"
`;

exports[`http invalid headers type 1`] = `
"main.test1.req.headers: should be object value at line 5 col 16
"
`;

exports[`http invalid method type 1`] = `
"main.test1.req.method: should be string value at line 5 col 15
"
`;

exports[`http invalid method value 1`] = `
"main.test1.req.method: is not valid http method at line 5 col 15
"
`;

exports[`http invalid params miss 1`] = `
"main.test1.req: must have url params id
"
`;

exports[`http invalid query prop type 1`] = `
"main.test1.req.query.a: should be scalar value at line 6 col 12
"
`;

exports[`http invalid query type 1`] = `
"main.test1.req.query: should be object value at line 5 col 14
"
`;

exports[`http invalid req value 1`] = `
"main.test1.req: should be object value at line 3 col 10
"
`;

exports[`http invalid res headers prop type 1`] = `
"main.test1.res.headers.key: should be header value at line 8 col 14
"
`;

exports[`http invalid res headers type 1`] = `
"main.test1.res.headers: should be object value at line 7 col 16
"
`;

exports[`http invalid res status type 1`] = `
"main.test1.res.status: should be integer value at line 7 col 15
"
`;

exports[`http invalid res type 1`] = `
"main.test1.res: should be object value at line 6 col 10
"
`;

exports[`http invalid url miss 1`] = `
"main.test1.req.url: is required at line 3 col 10
"
`;

exports[`http invalid url type 1`] = `
"main.test1.req.url: should be string value at line 4 col 12
"
`;
